packages/ns-api/files/ns.report

```diff
@@ -315,13 +316,82 @@ def ovpnrw_bytes_by_hour_and_user(instance, day, user):
     return {"hours": hours_bytes}
 
 
+def fetch_and_find_fping_keys():
+    url =  "http://127.0.0.1:19999/api/v1/charts"
+    with urllib.request.urlopen(url) as response:
+        data = response.read()
+        json_data = json.loads(data)
+
+    # find charts that start with 'fping'
+    if 'charts' in json_data:
+        matching_keys = [key for key in json_data['charts'].keys() if key.startswith('fping')]
+
+    return matching_keys
+
+
+def get_fping_hosts():
+    # read fping hosts from /etc/netdata/fping.conf
+    try:
+        with open("/etc/netdata/fping.conf", 'r') as fp:
+            line = fp.readline()
+            line = line[7:-2]
+            hosts = line.split(" ")
+            return hosts
+    except:
+       return []
+    
+
+def get_netdata_chart_data(chart_name):
+    ret = {"labels": [], "data": []}
+
+    # retrieve chart data from netdata
+    url = f'http://127.0.0.1:19999/api/v1/data?chart={chart_name}&after=-3600&points=180&options=abs'
+    try:
+        with urllib.request.urlopen(url) as fu:
+            data = json.loads(fu.read())
+    except:
+        return ret
+
+    for record in data["data"]:
+        # first element is the label
+        ret["labels"].append(record[0])
```

Comment:
Are you sure that labels are not inside `data['labels']` field?

Reply:
Yes we are referring to the chart label, i.e. the value on x axis (the timestamp); let's improve the comment for clarity